code/modules/admin/verbs/anonymousnames.dm

```diff
@@ -0,0 +1,67 @@
+/client/proc/anon_names()
+	set category = "Admin - Events"
+	set name = "Setup Anonymous Names"
+
+
+	if(SSticker.current_state > GAME_STATE_PREGAME)
+		to_chat(usr, "This option is currently only usable during pregame.")
+		return
+
+	if(SSticker.anonymousnames)
+		SSticker.anonymousnames = ANON_DISABLED
+		to_chat(usr, "Disabled anonymous names.")
+		message_admins("<span class='adminnotice'>[key_name_admin(usr)] has disabled anonymous names.</span>")
+		return
+	var/list/names = list("Cancel", ANON_RANDOMNAMES, ANON_EMPLOYEENAMES)
+	var/result = input(usr, "Choose an anonymous theme","going dark") as null|anything in names
+	if(!usr || !result || result == "Cancel")
+		return
+	if(SSticker.current_state > GAME_STATE_PREGAME)
+		to_chat(usr, "You took too long! The game has started.")
+		return
+
+	SSticker.anonymousnames = result
+	to_chat(usr, "Enabled anonymous names. THEME: [SSticker.anonymousnames].")
+	message_admins("<span class='adminnotice'>[key_name_admin(usr)] has enabled anonymous names. THEME: [SSticker.anonymousnames].</span>")
+
+/**
+  * anonymous_name: generates a corporate random name. used in admin event tool anonymous names
+  *
+  * first letter is always a letter
+  * Example name = "Employee Q5460Z"
+  * Arguments:
+  * * M - mob for preferences and gender
+  */
+
```

Comment:
remove newlines between commennt and thing commented or else dmdoc doesnt work!!!!